### DISCUSSION TITLE
Live Graphing question

### DISCUSSION THREAD

**Discussion by tommiehall** [Live-Charts/LiveCharts2#598] [General]
Hello,  

I'm attempting to add a cartesian chart to a Winforms project that will update as data comes in over a serial port.    I have my communications working and the Com class currently updates my properties object in another class called Globals.    All of my other forms reference the Global class to pull the values of the properties.    They're currently all public float values.   There are 4 instances of each object that are stored in a bindinglist.     What would be the best way to set up a chart that updates when one of these properties are updated by the Com class?        Any information is welcomed,    including basic suggestions for an architecture.   I think I'm making it harder than it needs to be and just need a push in the right direction.

**Comment by grewegreg** [comment]
Have a look at this sample:

https://www.lvcharts.com/docs/WinForms/2.0.0-beta.330/samples.bars.race

Adding the [ObservableProperty] attribute allows the chart to be updated when the data is updated.  In the sample, the UpdateViewModel() method updates the data, which updates the chart.
